src/tests/readytorun/JittedMethodsCountingTest/JittedMethodsCountingTest.cs

```diff
@@ -48,4 +50,10 @@ private static bool IsHardwareIntrinsicsEnabled()
         return (string.IsNullOrEmpty(dotnetEnableHWIntrinsics)
                 || dotnetEnableHWIntrinsics != "0");
     }
+
+    private static bool IsSSE2Enabled()
+    {
+        string? dotnetSSE2 = Environment.GetEnvironmentVariable("DOTNET_EnableSSE2");
```

Comment:
I'm wondering whether we shouldn't rather use some runtime check to dynamically ascertain the availability of the instruction set extension but perhaps it's not much of a concern.

Reply:
If it's just about explicit suppression of SSE2 via the environment variable, then it makes perfect sense.

Reply:
Hmmm... That might be something worth exploring if it would make this simpler. Basically, the test should not run under circumstances where the JIT behaves differently than normal. Like for example, right now any of these are enough to make the JIT work differently and therefore we should skip the test:

- `DOTNET_ReadyToRun=0`
- `DOTNET_TieredCompilation=1`
- `DOTNET_EnableHWIntrinsic=0`
- `DOTNET_EnableSSE=0`
- `DOTNET_EnableSSE2=0`